aten/src/ATen/detail/AcceleratorHooksInterface.h

```diff
@@ -58,6 +58,10 @@ struct TORCH_API AcceleratorHooksInterface {
   virtual Device getDeviceFromPtr(void* data) const {
     TORCH_CHECK(false, "Backend doesn't support getDeviceFromPtr()");
   }
+
+  virtual Allocator* getDeviceAllocator() const {
```

Comment:
We need to add this for all accelerators to make sure we don't break anything right?
From the list in context.h, I guess we also need xpu, mtia, hpu and hip?